src/components/visualization/vegaLiteViz/vegaLiteViz.tsx

```diff
@@ -12,26 +13,50 @@ import type { VegaLiteData } from '../../types'
 function VegaLiteViz(props: VegaLiteData) {
   const chartRef = useRef<HTMLDivElement>(null)
   const [hasError, setHasError] = useState<boolean>(false)
-  const isDarkTheme = useTheme().palette.mode === 'dark'
+  const rusticTheme: Theme = useTheme()
+  const isDarkTheme = rusticTheme.palette.mode === 'dark'
+  const defaultFont = rusticTheme.typography.fontFamily
 
-  useEffect(() => {
-    function renderChart() {
-      if (chartRef.current && props.spec) {
-        const options = {
-          ...props.options,
-          theme: isDarkTheme ? props.theme?.dark : props.theme?.light,
+  function renderChart() {
+    if (chartRef.current && props.spec) {
+      const defaultConfig = {
+        font: defaultFont,
+      }
+
+      const { config, ...restOptions } = props.options || {}
+
+      let combinedConfig
+      if (config) {
+        if (typeof config === 'string') {
+          combinedConfig = config
+        } else {
+          combinedConfig = {
+            ...defaultConfig,
+            ...config,
+            font: config.font || defaultConfig.font,
+          }
         }
+      } else {
+        combinedConfig = defaultConfig
+      }
```

Comment:
could this be simplified ? maybe something like this -
```suggestion
      if (props.options?.config && typeof props.options.config !== 'string'){
         combinedConfig = {...props.options.config, ...defaultFontConfig}
       }
```

Reply:
I also wants to set default font when there's no config/options provided and this method won't cover that scenario. I've pushed my changes. Let me know if you have any feedback